node/src/actors/chain_manager/mining.rs

```diff
@@ -368,8 +374,22 @@ impl ChainManager {
                 .map(|(dr_pointer, reveals, dr_state)| {
                     debug!("Building tally for data request {}", dr_pointer);
 
-                    let results: Vec<Vec<u8>> =
-                        reveals.iter().map(|r| r.body.reveal.clone()).collect();
+                    // Ignore reveals that could not be decoded.
+                    //  Those reveal that cannot be decoded are most likely malformed and therefore
+                    //  their authors are punished in the same way as non-revealers.
+                    let results: Vec<RadonTypes> = reveals.clone()
+                        .into_par_iter()
+                        .filter_map(|reveal_tx| {
+                            let slice = reveal_tx.body.reveal.as_slice();
+                            match RadonTypes::try_from(slice) {
```

Comment:
This logic should be inside `RadManager`. Maybe the `RunTally` message can accept the input as `Vec<u8>` and do the decoding?